homeassistant/components/lightwave/climate.py

```diff
@@ -0,0 +1,173 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Find and return LightWave lights."""
+    if not discovery_info:
+        return
+
+    trv = []
+    lwlink = hass.data[LIGHTWAVE_LINK]
+    trv_proxy_ip = hass.data[LIGHTWAVE_TRV_PROXY]
+    trv_proxy_port = hass.data[LIGHTWAVE_TRV_PROXY_PORT]
+
+    for device_id, device_config in discovery_info.items():
+        name = device_config[CONF_NAME]
+        serial = device_config[CONF_SERIAL]
+        trv.append(
+            LightwaveTrv(name, device_id, lwlink, serial, trv_proxy_ip, trv_proxy_port)
+        )
+
+    async_add_entities(trv)
+
+
+class LightwaveTrv(ClimateDevice):
+    """Representation of a LightWaveRF TRV."""
+
+    def __init__(self, name, device_id, lwlink, serial, trv_proxy_ip, trv_proxy_port):
+        """Initialize LightwaveTrv entity."""
+        self._name = name
+        self._device_id = device_id
+        self._state = None
+        self._max_temp = DEFAULT_MAX_TEMP
+        self._min_temp = DEFAULT_MIN_TEMP
+        self._current_temperature = None
+        self._target_temperature = None
+        self._temperature_unit = TEMP_CELSIUS
+        self._target_temperature_step = 0.5
+        self._hvac_mode = HVAC_MODE_HEAT
+        self._hvac_action = None
+        self._lwlink = lwlink
+        self._battery = None
+        self._serial = serial
+        self._proxy_ip = trv_proxy_ip
+        self._proxy_port = trv_proxy_port
+        self._inhibit = 0
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    def update(self):
+        """Communicate with a Lightwave RTF Proxy to get state."""
+        targ = temp = battery = trv_output = None
+        (temp, targ, battery, trv_output) = self._lwlink.read_trv_status(self._serial)
+        if temp is not None:
+            self._current_temperature = temp
+        if targ is not None:
+            if self._inhibit == 0:
+                self._target_temperature = targ
+                if targ == 0:
+                    # TRV off
+                    self._target_temperature = None
+                if targ >= 40:
+                    # Call for heat mode, or TRV in a fixed position
+                    self._target_temperature = None
+            else:
+                # Done the job - use proxy next iteration
+                self._inhibit = 0
+        if battery is not None:
+            self._battery = battery
+        if trv_output is not None:
+            if trv_output > 0:
+                self._hvac_action = CURRENT_HVAC_HEAT
+            else:
+                self._hvac_action = CURRENT_HVAC_OFF
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return {
+            ATTR_BATTERY_LEVEL: self._battery,
+            "Device Type": "LightwaveRF TRV",
```

Comment:
Use snake case for attributes:
```suggestion
            "device_type": "LightwaveRF TRV",
```